src/commands/up.rs

```diff
@@ -187,10 +187,20 @@ impl UpCommand {
                 cmd.arg("--help-args-only");
             }
             TriggerExecOpts::Local { app, working_dir } => {
-                let locked_app = spin_trigger::locked::build_locked_app(app, &working_dir)?;
+                let locked_app = spin_trigger::locked::build_locked_app(app.clone(), &working_dir)?;
+
+                let state_dir = match app.info.origin {
+                    ApplicationOrigin::File(mut f) => {
+                        f.pop();
+                        f
```

Comment:
Could this not just be `ApplicationOrigin::File(f) => f.parent`?  The mutation, and resultant need for statement flow rather than an expression, are a bit jarring to me.  (Subjective of course.)

Reply:
updated.